InMobiAdapter/src/main/java/com/chartboost/mediation/inmobiadapter/InMobiAdapter.kt

```diff
@@ -512,16 +536,30 @@ class InMobiAdapter : PartnerAdapter {
                 return Result.failure(ChartboostMediationAdException(ChartboostMediationError.CM_LOAD_FAILURE_INVALID_PARTNER_PLACEMENT))
             }
 
-            return suspendCoroutine { continuation ->
-                InMobiInterstitial(
+            return suspendCancellableCoroutine { continuation ->
+                inMobiInterstitialAds[request.identifier] = InMobiInterstitial(
```

Comment:
did this work? We're already storing InMobiInterstitial as part of `partnerAd` which is held by ChartboostMediationFullscreenAd or something of that sort.
what we aren't storing is the listener from buildFullscreenAdListener.
although if this is what it takes, then sure, way easier!

Reply:
Yep, current commit eb5852ff954d55dbac3f4da9d3f6c8133009fae1 is working and is not experiencing the issue described.

Will make a few enhancements on the next commits.